app/src/main/proto/vss.proto

```diff
@@ -98,6 +98,64 @@ message PutObjectRequest {
 message PutObjectResponse {
 }
 
+message GetKeysSummaryRequest {
+
+  // store_id is a keyspace identifier.
+  // Ref: https://en.wikipedia.org/wiki/Keyspace_(distributed_data_store)
+  // All APIs operate within a single store_id.
+  // It is up to clients to use single or multiple stores for their use-case.
+  // This can be used for client-isolation/ rate-limiting / throttling on the server-side.
+  // Authorization and billing can also be performed at the store_id level.
+  string store_id = 1;
+
+  // A key_prefix is a string of characters at the beginning of the key. Prefixes can be used as
+  // a way to organize key-values in a similar way to directories.
+  //
+  // If key_prefix is specified, the response results will be limited to those keys that begin with
+  // the specified prefix.
+  //
+  // If no key_prefix is specified or it is empty, all the keys are eligible to be returned in the
+  // response.
+  optional string key_prefix = 2;
+
+  // last_evaluated_key is a pagination token.
+  //
+  // To query for the first page of GetKeysSummary, last_evaluated_key must not be specified.
+  //
+  // For subsequent pages, use the value that was returned as `last_evaluated_key` in the previous
+  // page's GetKeysSummaryResponse.
+  // It will be used as exclusive_start_key to perform the new query and only keys greater than
+  // the specified last_evaluated_key will be returned in the response.
+  optional string last_evaluated_key = 3;
+}
+
+message GetKeysSummaryResponse {
+
+  // Fetched versions along with the corresponding keys in the request.
+  repeated KeyValue key_versions = 1;
+
+  // last_evaluated_key is a pagination token, it is the key of the item where the operation stopped,
+  // inclusive of the current result set.
+  // Use this value to start a new GetKeysSummary operation, excluding this value in the new result
+  // by specifying this last_evaluated_key in the next request.
+  //
+  // If last_evaluated_key is empty, then the "last page" of results has been processed and there is
```

Comment:
Added some clarification.

Since this is a response object, we want to be as specific as possible, so that it is easier for client to understand. I don't want client to have to think whether it will be empty or not set (even though they are kinda same in protobuf) 

Whereas in request structs, we want to be more forgiving of client's usage, both empty and "not set" serve the same purpose and even if client uses one or the other, things should just work and server will handle it.